### PR TITLE
Init Codecov.

### PR DESCRIPTION
**Pull request by xuezhaojun**

### CODE REVIEW

File: .github/workflows/go-presubmit.yml [hunk 1]
```diff
@@ -26,9 +23,6 @@ jobs:
     steps:
       - name: checkout code
         uses: actions/checkout@v2
-        with:
-          fetch-depth: 1
-          path: go/src/open-cluster-management.io/cluster-proxy
```
Comment (qiujian16) [inline .github/workflows/go-presubmit.yml:31]: why we need to remove this field?

Comment (xuezhaojun) [reply .github/workflows/go-presubmit.yml:31]: For 2 reasons:
1. Simplicity, it doesn't harm anything if we remove it.
2. Codecov needs `cover.out` to have the same file path as files in the GitHub repo. (We can config Codecov but it's more simple to remove the above thing)

**Pull request merged**